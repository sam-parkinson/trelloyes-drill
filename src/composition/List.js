import React from 'react';
import Card from './Card';
import './List.css';

class List extends React.Component {
  static defaultProps = {
    onClickAdd: () => {},
  }

  makeCards = () => {
    const cardDiv = this.props.cards.map(i => (
      <Card
        key={i.id}
        id={i.id}
        title={i.title}
        content={i.content}
        deleteItem={this.props.deleteItem}
      >
      </Card>
      ) 
    )
    return cardDiv
  }

  render() {
    return (
      <section className='List' key={this.props.id}>
        <header className='List-header'>
          <h2>{this.props.header}</h2>
        </header>
        <div className='List-cards'>
          {this.makeCards()}
          <button 
            onClick={() => this.props.addItem(this.props.id)}
            key={this.props.id}
          >
            + Add Random Item
          </button>
        </div>
      </section>
    );
  }
}

export default List