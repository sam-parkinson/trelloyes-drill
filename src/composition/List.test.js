import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

const cards = [{
  title: 'test',
  content: 'test',
  id: 'z'
}];

describe('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List className='List' cards={cards} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List className='List' cards={cards} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});