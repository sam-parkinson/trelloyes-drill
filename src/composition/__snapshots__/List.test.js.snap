// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        test
      </h3>
      <p>
        test
      </p>
    </div>
  </div>
</section>
`;
